scripts/horton-regression-test.py

```diff
@@ -0,0 +1,318 @@
+    # the force option is used.
+    rt_new_previous = rt_previous.copy()
+    changed = False
+    for key, status in rt_status.iteritems():
+        if status != 'OK' or do_force:
+            print '   Updating "{}"'.format(key)
+            rt_new_previous[key] = rt_results[key]
+            changed = True
+        else:
+            print '   Not updating "{}"'.format(key)
+
+    if changed:
+        print '   Rewriting example code.'
+        # Load the example code.
+        with open(example_path) as f:
+            example_lines = f.readlines()
+
+        # Search for the autogenerated previous results and strip them.
+        try:
+            index_auto = example_lines.index(comment_auto)
+        except ValueError:
+            index_auto = None
+        if index_auto is not None:
+            example_lines = example_lines[:index_auto]
+
+        # Add new previous results
+        example_lines.append(comment_auto)
+        if any(isinstance(x, np.ndarray) for x in rt_new_previous.itervalues()) and \
+           'import numpy as np\n' not in example_lines:
+            example_lines.append('import numpy as np  # pylint: disable=wrong-import-position\n')
+        example_lines.append('rt_previous = {\n')
+        for key, new_previous in rt_new_previous.iteritems():
+            representation = format_value(new_previous, key, example_path)
+            example_lines.append('    \'{}\': {},\n'.format(key, representation))
+        example_lines.append('}\n')
+
+        # Write the updated code.
+        with open(example_path, 'w') as f:
+            f.writelines(example_lines)
+
+    # Write basic unit test code to screen.
+    print template_unit_test.format(
+        example_name=example_path[14:-3].replace('/', '_'),
+        example_path=example_path[5:],
+        example_basename=os.path.basename(example_path[5:]),
+        example_dirname=os.path.dirname(example_path[5:]),
+    )
+
+
+def format_value(value, key, example_path):
+    """Format a previous value in full precision."""
+    if isinstance(value, np.ndarray):
+        # Multidimensional arrays are written as raveled vectors and are later reshaped
+        # properly. This facilitates code formatting and results in more compact code.
+        result = 'np.array({})'.format(value.ravel().tolist())
+        if len(value.shape) > 1:
+            result += '.reshape{}'.format(value.shape)
+        if len(result) > 82 - len(key):
+            # Wrap array over multiple lines, assuming key is not excessively long.
+            # The result must be PEP8 compatible.
+            data = ', '.join(repr(v) for v in value.flat)
+            lines = textwrap.wrap(data, 90, initial_indent=' '*8, subsequent_indent=' '*8)
+            lines = ['np.array(['] + lines + ['    ])']
+            result = '\n'.join(lines)
+            if len(value.shape) > 1:
+                result += '.reshape{}'.format(value.shape)
```

Comment:
The string representation is redone from scratch when the one-line format is too long. There should be a nicer way to do this.